src/RedLock.php

```diff
@@ -72,10 +72,17 @@ public function unlock(array $lock)
 
     private function initInstances()
     {
+        $app = app();
+        // support newer and older Laravel 5.*
+        if (method_exists($app, 'makeWith')) {
+            $makeFunc = 'makeWith';
+        } else {
+            $makeFunc = 'make';
+        }
         if (empty($this->instances)) {
             foreach ($this->servers as $server) {
                 //list($host, $port, $database) = $server;
-                $redis = App::make(Redis::class, [$server]);
+                $redis = $app->$makeFunc(Redis::class, [$server]);
```

Comment:
you need to pass the parameter name here:
`$redis = App::makeWith(Redis::class, ['parameters' => $server]);`

`$redis = App::make(Redis::class, [$server]);` This will also work but only on localhost because while calling like that parameters will not be passed to Redis client and it will choose the default localhost values everytime

This is the prototype of predis constructor where we have to pass $parameters.
`public function __construct($parameters = null, $options = null)`

one of the same issue:
https://github.com/laravel/framework/issues/21147

Reply:
something like this:
```private function initInstances()
      {
         $app = app();
         if (empty($this->instances)) {
              foreach ($this->servers as $server) {
                  //list($host, $port, $database) = $server;
                 if ($app->version() >= 5.4) {
                 $redis = App::makeWith(Redis::class, ['parameters' => $server]);
                 }
                 else {
                  $redis = App::make(Redis::class, [$server]);
                  }
                //$redis->connect($host, $port, $timeout);
                  $this->instances[] = $redis;
              }```

Reply:
Thanks, I'll look into that

Reply:
I changed that, thanks again.

Reply:
no problem